server/controllers/orderController.ts

```diff
@@ -15,4 +15,37 @@ export default class OrderController {
       order,
     })
   }
+
+  confirmDelete: RequestHandler = async (req: Request, res: Response) => {
+    const id = req.params.orderId
+    const order = await this.orderService.getOrder(id)
+
+    if (order.status === 'Submitted') {
+      res.redirect('/order/delete/failed')
+    } else {
+      res.render('pages/order/delete-confirm', {
+        order,
+      })
+    }
+  }
+
+  delete: RequestHandler = async (req: Request, res: Response) => {
+    const id = req.params.orderId
+    const order = await this.orderService.getOrder(id)
+
+    if (order.status === 'Submitted') {
+      res.redirect('/order/delete/failed')
+    } else {
+      await this.orderService.deleteOrder(id)
```

Comment:
So at the minimum if there is an error is `orderServicedeleteOrder`, we will show the default error page. There isn't a specific design for if the deletion fails so for now this will do I think.